pkg/config/config.go

```diff
@@ -0,0 +1,303 @@
+	//
+	// "Never"
+	//   Never pulls an image, i.e. only uses a local image.
+	//   Container will fail if the image isn't present
+	//
+	// "IfNotPresent"
+	//   Pulls if the image isn't present on disk.
+	//   Container will fail if the image isn't present and the pull fails.
+	//
+	// Added in v0.8.0.
+	ImagePullPolicy v1.PullPolicy
+}
+
+// HTTPConfig holds settings for the HTTP server.
+type HTTPConfig struct {
+	CORS CORSConfig
+	// BindAddress is the IP-address and port, separated by a colon, to bind
+	// the HTTP server to. An IP-address of 0.0.0.0 will bind to all
+	// IP-addresses.
+	//
+	// Added in v0.8.0.
+	BindAddress string
+}
+
+// CORSConfig holds settings for the HTTP server's CORS settings.
+type CORSConfig struct {
+	// AllowAllOrigins enables CORS and allows all hostnames and URLs in the
+	// HTTP request origins when set to true. Practically speaking, this
+	// results in the HTTP header "Access-Control-Allow-Origin" set to "*".
+	//
+	// Added in v0.8.0.
+	AllowAllOrigins bool
+	// AllowOrigins enables CORS and allows the list of origins in the
+	// HTTP request origins when set. Practically speaking, this
+	// results in the HTTP header "Access-Control-Allow-Origin".
+	//
+	// Added in v0.8.0.
+	AllowOrigins []string
+}
+
+// AggregatorConfig holds settings for the aggregator.
+type AggregatorConfig struct {
+	// WharfAPIURL is the URL used to connect to Wharf API.
+	//
+	// Added in v0.8.0.
+	WharfAPIURL string
+	// WorkerAPIExternalPort is the port used to connect to a Wharf worker.
+	//
+	// Added in v0.8.0.
+	WorkerAPIExternalPort string
+}
+
+// WatchdogConfig holds settings for the watchdog.
+type WatchdogConfig struct {
+	// WharfAPIURL is the URL used to connect to Wharf API.
+	//
+	// Added in v0.8.0.
+	WharfAPIURL string
+	// ProvisionerURL is the URL used to connect to the Wharf Cmd
+	// provisioner.
+	//
+	// Added in v0.8.0.
+	ProvisionerURL string
+}
+
+// DefaultConfig is the hard-coded default values for wharf-cmd's configs.
+var DefaultConfig = Config{
+	K8s: K8sConfig{
+		Context:   "",
+		Namespace: "",
+	},
+	Worker: WorkerConfig{
+		Steps: StepsConfig{
+			Docker: DockerStepConfig{
+				Image: "gcr.io/kaniko-project/executor:v1.7.0",
+			},
+			Kubectl: KubectlStepConfig{
+				Image: "docker.io/wharfse/kubectl:v1.23.5",
+			},
+			Helm: HelmStepConfig{
+				Image: "docker.io/wharfse/helm:v3.8.1",
+			},
+		},
+	},
+	Provisioner: ProvisionerConfig{
+		HTTP: HTTPConfig{
+			CORS: CORSConfig{
+				AllowAllOrigins: false,
+				AllowOrigins:    []string{},
+			},
+			BindAddress: "0.0.0.0:5009",
+		},
+		Worker: ProvisionerWorkerConfig{
+			ServiceAccountName: "wharf-cmd",
+			InitContainer: K8sContainerConfig{
+				Image:           "bitnami/git",
+				ImageTag:        "2-debian-10",
+				ImagePullPolicy: v1.PullIfNotPresent,
+			},
+			Container: K8sContainerConfig{
+				Image:           "quay.io/iver-wharf/wharf-cmd",
+				ImageTag:        "latest",
+				ImagePullPolicy: v1.PullAlways,
+			},
+		},
```

Comment:
These `provisioner.worker.*` configs are strictly k8s-related. I think they should have a `k8s` part in there, but I'm uncertain about either:

- `provisioner.worker.k8s.*`
- `provisioner.k8s.worker.*`

I'm thinking the latter, `provisioner.k8s.worker.*`, as that makes more sense to me where additional provisioner &times; kubernetes configs would be grouped together then.

We do want them separate though, as we will have other container configs for Docker (such as no ImagePullPolicy).

Reply:
`provisioner.k8s.worker.*` seems good to me 👍🏻